File: Algorithm_LevelOne/Two_Sum_II_Input_Array_Is_Sorted/Two_Sum_II_Input_Array_Is_Sorted.py
# 167. Two Sum II - Input Array Is Sorted Python

class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        minimum = 0
        maximum = len(numbers)-1
        while True:
            if numbers[maximum] + numbers[minimum] > target:
                maximum -= 1
                continue
            elif numbers[maximum] + numbers[minimum] < target:
                minimum += 1
            else:
                return [minimum + 1, maximum + 1]
            

File: Algorithm_LevelOne/Two_Sum_II_Input_Array_Is_Sorted/Two_Sum_II_Input_Array_Is_Sorted.cc
// 167. Two Sum II - Input Array Is Sorted C++

#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& numbers, int target) {
        int max = numbers.size();
        int min = 1;
        std::vector<int> output;
        while (true) {
            if (numbers[max-1] + numbers[min-1] > target) max -= 1;
            else if (numbers[max-1] + numbers[min-1] < target) min += 1;
            else {
                output.emplace_back(min);
                output.emplace_back(max);
                break;
            }
        }
        return output;
    }
};
